### PR TITLE
[TD]Detail View Fixes

### PR DESCRIPTION
**Pull request by WandererFan**
This PR is a collection of fixes for the Detail view.  The biggest change is to fix issue 4486 detail view of non-solids.   There are also small fixes for matting size, a bad preference path and avoiding a segfault.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5471">pipeline status</a> for feature branch [PR_5471](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5471). Pipeline [460324979](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/460324979) was triggered at [5e529c5](https://github.com/FreeCAD/FreeCAD/pull/5471/commits/5e529c58abe446554cb5d812cb3c0619240b21e3). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5471">pipeline status</a> for feature branch [PR_5471](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5471). Pipeline [460325604](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/460325604) was triggered at [51a028f](https://github.com/FreeCAD/FreeCAD/pull/5471/commits/51a028f76860d5229728545f5d25301ea5edc1fe). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/Gui/DlgPrefsTechDrawAdvanced.ui [hunk 1]
```diff
@@ -599,8 +599,6 @@ Each unit is approx. 0.1 mm wide</string>
    <header>Gui/PrefWidgets.h</header>
   </customwidget>
  </customwidgets>
- <resources>
-  <include location="Resources/TechDraw.qrc"/>
```
Comment (donovaly) [inline src/Mod/TechDraw/Gui/DlgPrefsTechDrawAdvanced.ui:603]: Why this deletion?

By the way, due to Werner's recent changes for PrefQuantitySpinBox one maybe needs to recompute the Qt designer plugin. TD does not yet use use PrefQuantitySpinBox but in future you might want to use it.

**Comment by WandererFan** [comment]
I think Qt Designer must remove it if the ui doesn't pull anything (like an
icon) from the resource file.

Thanks for the tip on the plugins.  I will rebuild the library.

**Pull request merged**